packages/docs/src/components/DocSiteNavigation.tsx

```diff
@@ -123,57 +132,117 @@ const DocSiteNavigation = ({ location }: DocSiteNavProps) => {
     return {
       label: formatNavItemLabel(name),
       url,
-      id: url,
+      id: relativePath,
       selected: isItemSelected(url),
+      order: frontmatter.order || 0,
     };
   };
 
-  const formatNavData = (dataList: GraphQlNavItem[]): VerticalNavItemProps[] => {
-    const retVal: VerticalNavItemProps[] = [];
+  /**
+   * Need to nest level2 items that have subnav items under the relevant level1 items
+   */
+  const nestSubNavs = (dataList): VerticalNavItemProps[] => {
+    const level1Items: { string?: VerticalNavItemProps } = {};
+    const level2Items = [];
+
+    // sort level1 vs level2 items
     dataList.forEach((dataItem) => {
-      // for each level 1 item that has sub nav items
-      if (dataItem.fieldValue.length) {
-        // format all the level 2 items
-        const subNavItems = dataItem.edges.map((subNavItem) => formatNavItemData(subNavItem.node));
-
-        const labelText = formatNavItemLabel(dataItem.fieldValue);
-        const isSelected = isItemSelected(dataItem.fieldValue);
-        // add level 1 item & sub items
-        retVal.push({
-          label: labelText,
-          items: subNavItems,
-          defaultCollapsed: !isSubNavItemSelected(subNavItems),
-          selected: isSelected,
-        });
+      if (dataItem.label.split('/').length > 1) {
+        level2Items.push(dataItem);
       } else {
-        // for each level 1 item without sub nav items,
-        // add each top level item
-        dataItem.edges.forEach((navItemLvl1) => {
-          retVal.push(formatNavItemData(navItemLvl1.node));
-        });
+        level1Items[dataItem.label] = dataItem;
       }
     });
 
+    // iterate through level2 items and nest them under their level1 parent
+    level2Items.forEach((level2Item) => {
+      // sort level3 items by order then id / name
+      level2Item.items = sortBy(level2Item.items, ['order', 'id']);
+
+      const [level1Name, level2Name] = level2Item.label.split('/');
+
+      level1Items[level1Name].items.push({ ...level2Item, label: level2Name });
+
+      if (!level2Item.defaultCollapsed) {
+        // if any level2 item is default expanded, make sure that level1 item also gets default expanded
+        level1Items[level1Name].defaultCollapsed = level2Item.defaultCollapsed;
+      }
+    });
+
+    return Object.values(level1Items);
+  };
+
+  /**
+   * sort items - needed because level2 items that have sub nav are added to the end of list in previous step
+   */
+  const sortNavItems = (level1Items: VerticalNavItemProps[]): VerticalNavItemProps[] => {
+    level1Items.forEach((level1Item) => {
+      // sort items based on the order defined in frontmatter then id which is the file path for the item
+      level1Item.items = sortBy(level1Item.items, ['order', 'id']);
+    });
+
+    level1Items.sort((itemA, itemB) => {
+      return level1ItemOrder.indexOf(itemA.label) - level1ItemOrder.indexOf(itemB.label);
+    });
+    return level1Items;
+  };
```

Comment:
It might also be worth noting in the doc comments for the `restructureAsVerticalNavItem` that the level1 and level2 items (or all items that have sub-items) are all mixed together at that point and still need to be organized. That's what took me a while to understand.

Reply:
Done. I tried to update the docs and types a bit to be more descriptive